patia_adv_logger: Include readme in module docs (#1137)
OpenDevicePartnership/patina: components/patina_adv_logger/README.md

# Patina Advanced Logger

> The Patina Advanced Logger crate provides a [log::Log](https://crates.io/crates/log) implementation that provides

Reviewer: ```suggestion
The Patina Advanced Logger crate provides a [log::Log](https://crates.io/crates/log) implementation including

```

in-memory and serial logging capabilities and a component for producing the Advanced Logger protocol. The in-memory
logging is written to the advanced logger memory buffer, whose address is produced via the Advanced logger guided HOB

> described below. In addition to the runtime functionality described above, this crate also provides command line tool

Reviewer 2: ```suggestion
described below. In addition to the runtime functionality described above, this crate also provides a command line tool
```

to parse parse logs pulled off of a physical device, and a public parser object for creating custom tooling.

## Memory Log Behavior

The crate stores records in a shared buffer that begins with an `ADVANCED_LOGGER_INFO` header.

> - Aligned entries follow the header.

Reviewer: ```suggestion
- Aligned entries follow in the memory after the header.

```

use patina_adv_logger::{component::AdvancedLoggerComponent, logger::AdvancedLogger};

use log::LevelFilter;
use core::ffi::c_void;

static LOGGER: AdvancedLogger<UartNull> = AdvancedLogger::new(
   Format::Standard, // How logs are formatted
   &[("allocations", LevelFilter::Off)], // set custom log levels per module
   log::LevelFilter::Info, // Default log level
   UartNull { }, // Serial writer instance
);

struct ExamplePlatform;

impl ComponentInfo for ExamplePlatform {
   fn components(mut add: Add<Component>) {
      add.component(AdvancedLoggerComponent::<UartNull>::new(&LOGGER));
   }
}

#[cfg_attr(target_os = "uefi", unsafe(export_name = "efi_main"))]
pub extern "efiapi" fn _start(physical_hob_list: *const c_void) -> ! {
   log::set_logger(&LOGGER).map(|()| log::set_max_level(log::LevelFilter::Trace)).unwrap();

   // SAFETY: The physical_hob_list pointer is assumed to be valid as it is provided to the entry_point from an
   // external caller.
   if let Err(e) = unsafe { LOGGER.init(physical_hob_list) } {
      log::error!("Failed to find a the Advanced Logger HOB. Cannot write to the memory log.");
   }

   # loop { }
}
```

### Platform Integration

> Currently, the Patina Advanced Logger expects that a log buffer has already been created prior to the Patina DXE Core

Reviewer: ```suggestion
The Patina Advanced Logger expects that a log buffer has already been created prior to the Patina DXE Core

```

being executed. The location of this buffer is provided via a HOB in the HOB list. So long as it is provided, the

> logger and component will execute as expected. If your platform is a EDKII style platform, a [PEI Core Library](https://github.com/microsoft/mu_plus/blob/d0d305b620baced42adf16b2387af9412fdc0ef9/AdvLoggerPkg/Library/AdvancedLoggerLib/PeiCore/AdvancedLoggerLib.inf)

Reviewer 3: nit
```suggestion
logger and component will execute as expected. If your platform is a EDK II style platform, a [PEI Core Library](https://github.com/microsoft/mu_plus/blob/d0d305b620baced42adf16b2387af9412fdc0ef9/AdvLoggerPkg/Library/AdvancedLoggerLib/PeiCore/AdvancedLoggerLib.inf)
```

If it's easy enough to find and replace in these modified files, please do so. Otherwise, feel free to ignore.
